### PR TITLE
validate the changelog to catch errors before it's time to release

### PR DESCRIPTION
**Pull request by shiftkey**
Turns out I broke the changelog in b0966a27bd8815f0ae0f004d078557e38fe3db10 which affected publishing a release today. Oops, sorry @nerdneha!

This PR is an attempt to catch this issue and some others that would affect releases before they get merged.

 - [x] commit a malformed changelog JSON -> :boom: goes CI, helpful error message

 - [x] catch a duplicate changelog entry -> :boom: goes CI, helpful error message

 - [x] add a release that doesn't match our expected format -> :boom: goes CI, helpful error message

 - [x] fix these issues - ✅ goes CI

### CODE REVIEW

File: script/eslint.ts [hunk 1]
```diff
@@ -29,6 +30,6 @@ console.log(eslint.getFormatter()(report.results))
 if (report.errorCount > 0) {
   process.exitCode = 1
   console.error(
-    chalk`{bold.green → To fix some of these errors, run {underline yarn eslint:fix}}`
+    chalk`{bold.green → To fix some of these errors, run {underline yarn eslint --fix}}`
```
Comment (shiftkey) [inline script/eslint.ts]: Because:

File: changelog.json [hunk 2]
```diff
@@ -670,8 +670,7 @@
       "[Fixed] 'Create Without Pushing' button throwing an exception while opening a pull request - #2368",
       "[Fixed] Windows: Don't removing the running app out from under itself when there are updates pending - #2373",
       "[Fixed] Windows: Respect `core.autocrlf` and `core.safeclrf` when modifying the .gitignore - #1535",
-      "[Fixed] Windows: Fix opening the app from the command line - #2396",
-      "[Fixed] 'Create Without Pushing' button throwing an exception while opening a pull request - #2368"
+      "[Fixed] Windows: Fix opening the app from the command line - #2396"
```
Comment (iAmWillShepherd) [inline changelog.json]: Just curious why this change is here. Didn't this get fixed when we pushed the release a few days ago?

File: script/validate-changelog.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env ts-node
+
+import * as Path from 'path'
+import * as Fs from 'fs'
+
+import * as Ajv from 'ajv'
+
+function handleError(error: any) {
```
Comment (iAmWillShepherd) [inline script/validate-changelog.ts]: This can be typed as `string` correct?

File: changelog.json (hunk 2 above)
Comment (shiftkey) [reply changelog.json]: This was caught because we can validate that the array of strings in a release entry are unique:



I figured this would be helpful to catch duplicate entries from appearing in the changelog, but it's not something I've got strong feelings about.

File: script/validate-changelog.ts (hunk 3 above)
Comment (shiftkey) [reply script/validate-changelog.ts]: Sure can!

File: changelog.json (hunk 2 above)
Comment (iAmWillShepherd) [reply changelog.json]: oh, I see. I'm all good for keep this in 👍

**Comment by shiftkey** [comment]
:apple:

File: .travis.yml [hunk 4]
```diff
@@ -57,7 +57,7 @@ install:
   - yarn install --force
 
 script:
-  - yarn lint && yarn build:prod && yarn test:setup && yarn test
+  - yarn lint && yarn build:prod && yarn validate-changelog && yarn test:setup && yarn test
```
Comment (iAmWillShepherd) [inline .travis.yml]: I think we should validate the `change-log` first since it completes a lot faster than building. Nothing worse than waiting on CI for 15 mins just to find out our change-log is messed up.

Comment (iAmWillShepherd) [reply .travis.yml]: I want to clarify that I meant we should validate before building.

**Comment by PRATHEEP2521** [comment]
4336
